src/gui/tracks/TrackContentWidget.cpp

```diff
@@ -74,6 +77,16 @@ TrackContentWidget::TrackContentWidget( TrackView * parent ) :
 			SIGNAL( positionChanged( const lmms::TimePos& ) ),
 			this, SLOT( changePosition( const lmms::TimePos& ) ) );
 
+	// Update background if snap size changes
+	connect( getGUI()->songEditor()->m_editor->snappingModel(),
+			SIGNAL( dataChanged() ),
+			this, SLOT( updateBackground() ) );
+
+	// Also update background if proportional snap is enabled/disabled
+	connect( getGUI()->songEditor()->m_editor,
+			SIGNAL( proportionalSnapChanged() ),
+			this, SLOT( updateBackground() ) );
```

Comment:
We prefer to use newer function-pointer-based signals and slots, rather than the old macro-based ones. It would look something like this (not tested):
```cpp
	// Update background if snap size changes
	connect(getGUI()->songEditor()->m_editor->snappingModel(), &Model::dataChanged,
			this, &TrackContentWidget::updateBackground);

	// Also update background if proportional snap is enabled/disabled
	connect(getGUI()->songEditor()->m_editor, &SongEditor::proportionalSnapChanged,
			this, &TrackContentWidget::updateBackground);
```